bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/GatewayBridgeHandler.java

```diff
@@ -0,0 +1,325 @@
+     */
+    private List<ShadeThingHandler> getShadeThingHandlers() throws IllegalStateException {
+        logger.debug("getShadeThingHandlers()");
+        return getThing().getThings().stream().map(Thing::getHandler).filter(ShadeThingHandler.class::isInstance)
+                .map(ShadeThingHandler.class::cast).collect(Collectors.toList());
+    }
+
+    /**
+     * Getter for the webTargets.
+     *
+     * @return the webTargets.
+     * @throws IllegalStateException if webTargets is not initialized.
+     */
+    public GatewayWebTargets getWebTargets() throws IllegalStateException {
+        GatewayWebTargets webTargets = this.webTargets;
+        if (webTargets != null) {
+            return webTargets;
+        }
+        throw new IllegalStateException("WebTargets not initialized.");
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (RefreshType.REFRESH == command) {
+            scheduler.submit(() -> doRefresh());
+            return;
+        }
+        Channel channel = getThing().getChannel(channelUID.getId());
+        if (channel == null) {
+            return;
+        }
+        ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+        if (channelTypeUID == null) {
+            return;
+        }
+        if (channelTypeId.equals(channelTypeUID.getId()) && OnOffType.ON == command) {
+            try {
+                getWebTargets().activateScene(Integer.parseInt(channelUID.getIdWithoutGroup()));
+            } catch (HubProcessingException | IllegalStateException e) {
+                logger.warn("handleCommand() exception:{}, message:{}", e.getClass().getSimpleName(), e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        HDPowerViewHubConfiguration config = getConfigAs(HDPowerViewHubConfiguration.class);
+        String host = config.host;
+
+        if (host == null || host.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.no-host-address");
+            return;
+        }
+
+        webTargets = new GatewayWebTargets(this, httpClient, clientBuilder, eventSourceFactory, host);
+        scenesLoaded = false;
+        propertiesLoaded = false;
+        disposing = false;
+
+        /*
+         * Normally the thing's position state is updated by SSE. However we must do a refresh once on start up in order
+         * to get the initial state. Also the other properties (battery, signal strength etc.) are not updated by SSE.
+         * Furthermore we need to do periodic refreshes just in case the SSE connection may have been lost. So we
+         * schedule the refresh at the 'hardRefresh' interval.
+         */
+        ScheduledFuture<?> refreshTask = this.refreshTask;
+        if (refreshTask != null) {
+            refreshTask.cancel(false);
+        }
```

Comment:
This it not needed as already done by `dispose`. The framework will not call `initialize` two times without calling dispose in between.